alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -126,10 +127,11 @@ class TransportDeleteWorkflowAction @Inject constructor(
 
                 if (canDelete) {
                     val deleteResponse = deleteWorkflow(workflow)
```

Comment:
Should we attempt to delete monitors first and then delete the workflow? And only delete everything if the user has permission to delete everything?